src/compiler/compile/render_dom/wrappers/Element/index.ts

```diff
@@ -923,7 +923,7 @@ export default class ElementWrapper extends Wrapper {
 			const snippet = expression.manipulate(block);
 			let cached_snippet;
 			if (should_cache) {
-				cached_snippet = block.get_unique_name(`style_${name}`);
+				cached_snippet = block.get_unique_name(`style_${name}`.replace(/-/, '_'));
```

Comment:
Perhaps the replace should be done on `name` instead of the whole string? Not that it makes a significant difference but just for consistency with other similar occurrences

Reply:
It can't be since when svelte sets the styles a few lines later it won't set the correct styles. For example it will try and set `background_image` instead of `background-image`. We only need to change the variable since in JavaScript variables can't contain a `-`

Reply:
Also, this will only replace one hyphen

![image](https://user-images.githubusercontent.com/50981692/149120647-e6b0e794-689c-456b-b662-6085bedf4999.png)